storage/receipts.go

```diff
@@ -14,24 +18,19 @@ type ExecutionReceipts interface {
 	// ByID retrieves an execution receipt by its ID.
 	ByID(receiptID flow.Identifier) (*flow.ExecutionReceipt, error)
 
-	// Index indexes an execution receipt by block ID.
-	Index(blockID flow.Identifier, receiptID flow.Identifier) error
-
-	// ByBlockID retrieves an execution receipt by block ID.
-	ByBlockID(blockID flow.Identifier) (*flow.ExecutionReceipt, error)
-
-	// TODO: Index and ByBlockID assumes one block can only have one receipt,
-	// this is a valid assumption for execution node, and therefore these two
-	// functions can only be used by execution node.
-	// IndexByExecutor and ByBlockIDAllExecutionReceipts allows a block to have
-	// multiple receipts from different executors, and only stores one receipt per
-	// executor. This is from non-execution-node's perspective.
-	// We'd better split the above two use cases, and create two different receipts
-	// module to avoid misuse and confusion.
+	// ByBlockID retrieves all known execution receipts for the given block
+	// (from any Execution Node).
+	ByBlockID(blockID flow.Identifier) ([]*flow.ExecutionReceipt, error)
+}
 
-	// IndexByExecutor indexes an execution receipt by block ID and execution ID
-	IndexByExecutor(receipt *flow.ExecutionReceipt) error
+// MyExecutionReceipts holds and indexes Execution Receipts.
+// MyExecutionReceipts is an extension of storage.ExecutionReceipts API.
+// It introduces the notion of "MY execution receipt", from the viewpoint
+// of an individual Execution Node.
```

Comment:
Good idea and nice refactor 👍  It makes the difference more clear. 
Suggested to update the comments to explain why it's not an `extension` of the `storage.ExecutionReceipts` API
```suggestion
// MyExecutionReceipts reuses the storage.ExecutionReceipts API, but doesn't expose
// them. Instead, it includes the "My" in the method name in order to highlight the notion
// of "MY execution receipt", from the viewpoint of an individual Execution Node.
```